crypto/merklekeystore/keystore_test.go

```diff
@@ -17,176 +17,185 @@
+	a.Equal(1, length(signer, a))
 
-	_, err = New(1, 0, 1, crypto.Ed25519Type)
-	a.Error(err)
-
-	signer, err = New(2, 2, 2, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(1, len(signer.SignatureAlgorithms))
+	signer = generateTestSigner(2, 2, 2, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(1, length(signer, a))
 
 	sig, err := signer.Sign(genHashableForTest(), 2)
 	a.NoError(err)
 	a.NoError(signer.GetVerifier().Verify(2, 2, 2, genHashableForTest(), sig))
 
-	signer, err = New(2, 2, 3, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(0, len(signer.SignatureAlgorithms))
+	signer = generateTestSigner(2, 2, 3, a)
+	defer signer.keyStore.store.Close()
+
+	a.Equal(0, length(signer, a))
 
 	_, err = signer.Sign(genHashableForTest(), 2)
 	a.Error(err)
 
-	s, err := New(8, 21, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(8, 21, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(8, 20, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(8, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(10, 21, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(10, 21, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(10, 20, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(10, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(11, 20, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 1)
+	signer = generateTestSigner(11, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 1)
 
-	s, err = New(11, 19, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(0, len(s.SignatureAlgorithms))
+	signer = generateTestSigner(11, 19, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(0, length(signer, a))
 	_, err = signer.Sign(genHashableForTest(), 2)
 	a.Error(err)
 }
 func TestEmptyVerifier(t *testing.T) {
 	a := require.New(t)
 
-	signer, err := New(8, 9, 5, crypto.Ed25519Type)
-	a.NoError(err)
+	signer := generateTestSigner(8, 9, 5, a)
+	defer signer.keyStore.store.Close()
 	a.NotEqual(*signer.GetVerifier(), Verifier{})
-
 }
 func TestEmptySigner(t *testing.T) {
 	a := require.New(t)
+	var err error
 
 	h := genHashableForTest()
-	signer, err := New(8, 9, 5, crypto.Ed25519Type)
+	signer := generateTestSigner(8, 9, 5, a)
 	a.NoError(err)
-	a.Equal(0, len(signer.SignatureAlgorithms))
+	a.Equal(0, length(signer, a))
 
 	_, err = signer.Sign(h, 8)
 	a.Error(err)
 
 	_, err = signer.Sign(h, 9)
 	a.Error(err)
 
-	_, err = signer.Trim(10)
-	a.Error(err)
+	// should Trim return error when trying to trim above its last valid? or just drop what relevant keys he finds as it is currently?
```

Comment:
I think we should. consider this, Can you think of a normal flow in the code that we will want to remove more than what we have?

Reply:
I guess if the dilithium keys were not refreshed as quick as they should and we're past the last valid round.
Then the old keys deletion thread might just delete all keys up to that round